src/Service/SudoModeService.php

```diff
@@ -0,0 +1,51 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\SecurityExtensions\Service;
+
+use SilverStripe\Control\Session;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\ORM\FieldType\DBDatetime;
+
+class SudoModeService implements SudoModeServiceInterface
+{
+    use Configurable;
+
+    /**
+     * The lifetime that sudo mode authorization lasts for, in minutes
+     *
+     * @config
+     * @var int
+     */
+    private static $lifetime_minutes = 180;
+
+    /**
+     * The session key that is used to store the timestamp for when sudo mode was last activated
+     *
+     * @var string
+     */
+    private const SUDO_MODE_SESSION_KEY = 'sudo-mode-last-activated';
+
+    public function check(Session $session): bool
+    {
+        $lastActivated = $session->get(self::SUDO_MODE_SESSION_KEY);
+        // Not activated at all
+        if (!$lastActivated) {
+            return false;
+        }
+
+        // Activated within the last "lifetime" window
+        $nowTimestamp = DBDatetime::now()->getTimestamp();
+        return $lastActivated > ($nowTimestamp - $this->getLifetime() * 60);
```

Comment:
Does it make sense to "touch" (re-`activate`) the sudo mode when it's checked?

Reply:
No, that would invalidate the point of it